aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -234,17 +234,16 @@ void lshift_kernel(TensorIterator& iter) {
 }
 
 void logical_and_kernel(TensorIterator& iter) {
-  // We use if-else here specifically for bool instead of using iter.common_dtype() like the CUDA implementation because
-  // common_dtype() is unavailable for bfloat16.
```

Comment:
@mruberry You mentioned updating the kernels to use `common_dtype`. I see a comment here warning against using it with bfloat16, but I think it's outdated (mostly just calling it out). E.g. the two below examples which hit that path worked fine for me:
```
>>> a = torch.ones(2, dtype=torch.bfloat16)
>>> b = torch.ones(2, dtype=torch.float32)
>>> torch.logical_and(a, a)
tensor([True, True])
>>> torch.logical_and(a, b)
tensor([True, True])
```